### ISSUE TITLE
`NOT_AUTHORIZED` error when attempting to refund transactions via API

### ISSUE THREAD

**Issue by thomasplevy** [paypal/PayPal-REST-API-issues#301]
### General information

* Environment: Sandbox
* `PayPal-Debug-ID` values: `a4fea3ab45e44 `
* Language, language version, and OS: PHP 7.2.13-1+ubuntu14.04.1+deb.sury.org+1

--

### Issue description

When attempting to refund transactions via [`/v2/payments/captures/{capture_id}/refund`](https://developer.paypal.com/docs/api/payments/v2/#captures_refund) I'm met with `NOT_AUTHORIZED` errors.

```php
array (size=5)
  'name' => string 'NOT_AUTHORIZED' (length=14)
  'message' => string 'Authorization failed due to insufficient permissions.' (length=53)
  'debug_id' => string 'a4fea3ab45e44' (length=13)
  'details' => 
    array (size=1)
      0 => 
        array (size=5)
          'issue' => string 'PERMISSION_DENIED' (length=17)
          'field' => string 'capture_id' (length=10)
          'value' => string '72G40463FU720461G' (length=17)
          'description' => string 'You do not have permission to access or perform operations on this resource.' (length=76)
          'location' => string 'path' (length=4)
  'links' => 
    array (size=1)
      0 => 
        array (size=2)
          'href' => string 'https://developer.paypal.com/docs/api/v2/payments/#NOT_AUTHORIZED' (length=65)
```

I've done my very best to scour the internet for potential causes of this and I cannot find anything...

Thanks

**Comment by kslimani** [comment]
Sound like a credentials permissions issue according [documentation](https://developer.paypal.com/docs/api/payments/v2/#error-NOT_AUTHORIZED).

**Comment by thomasplevy** [comment]
@kslimani right... yea, thanks... how do you fix this?

I cannot find a single thing anywhere on what the proper permissions are to process a refund.

I see a single `grant_type` when getting the token (`client_credentials` and I don't see anything anywhere else on needing any special permissions or settings for this resource.

Thanks

**Comment by cjaentsch** [comment]
Same here... Did you find a solution yet?

**Comment by thomasplevy** [comment]
@cjaentsch 

Nope, still doesn't work

**Comment by cjaentsch** [comment]
> @cjaentsch
> 
> Nope, still doesn't work

Do we have the problem only in sandbox or also in live? (we haven't tested live env yet)

**Comment by thomasplevy** [comment]
@cjaentsch I haven't tested live (yet). Building an integration and haven't gotten anything stable enough to actually go live.

I figure if it doesn't work in sandbox I don't actually care if it works in live or not.

**Comment by cjaentsch** [comment]
We tested it today in live env, doesn't work either. Yet when using "v1/payments/capture/*/refund" it works! (note: "amount" is not optional, throws an internal error when not setting the amount config). Seems to be an issue with v2 only...

**Comment by nuno-findster** [comment]
I have the same problem... any solution?

**Comment by cjaentsch** [comment]
As mentioned above, use the v1 endpoint just for this request. All other requests can be v2.

**Comment by thomasplevy** [comment]
@cjaentsch not to be rude but are you an official maintainer here? I'd like an official answer. If the official answer is that this endpoint doesn't work as v2 the docs should be updated accordingly. Am I wrong?

**Comment by cjaentsch** [comment]
@thomasplevy No, I'm not an official maintainer. That's just what I found out to solve the problem for my projects.

**Comment by nemenos** [comment]
Same problem here, still not fixed?
I've lost lot of time trying to figure out what was happening

**Comment by thomasplevy** [comment]
Yea, it'd be really cool to even just have a maintainer weigh in on this even if it's just to confirm or disprove the report. Please!

**Comment by oceanapplications** [comment]
This is still ongoing.
